e2e/push/push.go

```diff
@@ -75,26 +76,24 @@ func (c *ctx) testPushCmd(t *testing.T) {
 			}
 			defer os.RemoveAll(tmpdir)
 
-			cmd, out, err := e2e.ImagePush(t, c.env.CmdPath, tt.imagePath, tt.dstURI)
-			switch {
-			case tt.expectSuccess && err == nil:
-				// PASS: expecting success, succeeded
-
-			case !tt.expectSuccess && err != nil:
-				// PASS: expecting failure, failed
-
-			case tt.expectSuccess && err != nil:
-				// FAIL: expecting success, failed
-
-				t.Logf("Running command:\n%s\nOutput:\n%s\n", cmd, out)
-				t.Errorf("unexpected failure: %v", err)
-
-			case !tt.expectSuccess && err == nil:
-				// FAIL: expecting failure, succeeded
+			args := tt.dstURI
```

Comment:
Can we make `args` a `[]string` and append to it so we dont need to split a string by a separator in `e2e.WithArgs()`?

Reply:
I had that question before and no it does not work. In a nutshell, the command is very sensitive to empty slots, meaning an element in the slice that would not be defined, which would make the command to fail even if everything looks fine (only a double space would appear when displaying the elements of the commands). I had very bad luck with unit tests, e2e tests and in the context of my MPI tool.
In other words, that is by far the safer way to deal with it and i did not manage to have another solution that never create a problem.

Reply:
I had that question before and the short answer is no: so far, i always end up with Go assuming that an element in the slice is empty, which will make the command fail (when displaying the args, the problem can be seen through a double space). So this is the only reliable way i know of dealing with it.

Reply:
Could you throw a comment on the code summarizing these comments so that others hopefully dont run into this issue when working with this file?

Reply:
Not handling an empty argument seems like a short coming. @gvallee Can you open an issue for that so we can make things more consistent across the board?

Reply:
Will do and as you can see I am having problems with Github, my comments appearing/disappearing from my view of the PR... :(